core/src/test/java/org/apache/calcite/test/ReflectiveSchemaTest.java

```diff
@@ -589,6 +592,14 @@ private void check(ResultSetMetaData metaData, String columnName,
         .returns("C=null\n");
   }
 
+  @Test public void testDivideDoubleBigDecimal() {
+    final CalciteAssert.AssertThat with =
+        CalciteAssert.that().withSchema("s", CATCHALL);
+    with.query("select \"wrapperDouble\" / \"bigDecimal\" as c\n"
+        + " from \"s\".\"everyTypes\"")
+        .runs();
```

Comment:
We can also give some cases: 'bigdecimal / null', 'null/bigdecimal', 'bigdecimal/double'.

Reply:
I find that `ReflectiveSchemaTest#testOp()` enumerates all type combinations for numeric operators ('+', '-', '*', '/'), including `null` content (e.g., `wrapperInteger`, `wrapperDouble`). I think it can cover these cases. So, maybe we can even remove this test case.
However, when I tested a `null` literal with the following test case.
```
  @Test public void testDivideDoubleBigDecimal() {
    final CalciteAssert.AssertThat with =
        CalciteAssert.that().withSchema("s", CATCHALL);
    with.query("select \"bigDecimal\" / null as c\n"
        + " from \"s\".\"everyTypes\"")
        .runs();
  }
```

It fails due to the **Assertion** below. When SqlBinaryOperator figuring out its `Monotonicity`, if the second operand is a Literal, it will cast the literal into BigDecimal. But the `Null` Literal can not be properly handled.
It seems to be the problem related with NULL literal, which demands a systematic solution.
Should we file a seperate Jira for this issue (it can also be trigger by 'wrapperLong'/null)?

```
java.lang.AssertionError: cannot cast null as class java.math.BigDecimal
	at org.apache.calcite.sql.SqlLiteral.getValueAs(SqlLiteral.java:351)
	at org.apache.calcite.sql.SqlCallBinding.getOperandLiteralValue(SqlCallBinding.java:217)
	at org.apache.calcite.sql.SqlBinaryOperator.getMonotonicity(SqlBinaryOperator.java:190)
	at org.apache.calcite.sql.SqlCall.getMonotonicity(SqlCall.java:182)
	at org.apache.calcite.sql.SqlCallBinding.getOperandMonotonicity(SqlCallBinding.java:188)
	at org.apache.calcite.sql.SqlAsOperator.getMonotonicity(SqlAsOperator.java:139)
	at org.apache.calcite.sql.SqlCall.getMonotonicity(SqlCall.java:182)
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertSelectList(SqlToRelConverter.java:3961)
```

Reply:
Well, this exception is not expected, personally, i think the proper place to throw is:

1. The type checker: if we do not allow null literal operands for DIVIDE
2. The execution runtime, if we do not support a `number/null` pattern expression(we should do it if we already allows the pattern in the type checker)

Fire another issue is a more proper way to go, thanks.